codegen/service/endpoint.go

```diff
@@ -258,7 +256,25 @@ func New{{ .VarName }}Endpoint(s {{ .ServiceVarName}}{{ range .Schemes }}, auth{
 			return nil, err
 		}
 {{- end }}
-{{- if .ResultRef }}
+{{- if .ViewedResult }}
+		res, view, err := s.{{ .VarName }}(ctx{{ if .PayloadRef }}, p{{ end }})
+		if err != nil {
+			return nil, err
+		}
+		vres := {{ .ViewedResult.ConvertToViewed.VarName }}(res)
```

Comment:
Looking at this now, it seems strange that we would create the viewed type to then project it right away. Why wouldn't we have view specific constructors instead so that we don't have to initialize fields that we nullify in the next call? Could we replace the constructor above with one constructor per view (and adjust the comment explaining how that works).

Reply:
Sure.. I will change that as well.